lib/svatok_codebreaker/game.rb

```diff
@@ -0,0 +1,153 @@
+Hint, Save.
+ '
+      about_game[:take_player_name] = 'Please, enter your name:'
+      about_game.values
+    end
+
+    def start
+      @secret_code = ''
+      4.times { @secret_code += rand(1..6).to_s }
+      '----------Game started. Enter guess:----------'
+    end
+
+    def codebreaker_answer(answer)
+      menu_command?(answer) ? menu_command(answer) : submit_guess(answer)
+    end
+
+    def menu_command?(command)
+      @menu_commands.include?(command)
+    end
+
+    def menu_command(command)
+      command.downcase!
+      send(('command_' + command).to_sym)
+    end
+
+    def method_missing(meth)
+      if meth.to_s =~ /^command_(\w+)$/
+        'You entered is not correct command. Enter the correct command.'
+      else
+        super
+      end
+    end
+
+    def command_exit
+      @exit_game = true
+      '----------Bye!----------'
+    end
+
+    def command_restart
+      @attempts = ATTEMPTS
+      @hint = true
+      old_secret_code = @secret_code
+      start while old_secret_code == @secret_code
+      '----------Game restarted. Enter guess:----------'
+    end
+
+    def command_hint
+      return 'You have already used a hint.' unless @hint
+      @hint = false
+      @secret_code.to_s[rand(0..3)]
+    end
+
+    def command_save
+      return 'Sorry, but command Save not available at the moment.' unless (@marking_guess == '++++' || @attempts.zero?)
+        score_file = File.open(@file_path, 'a')
+        score_file.puts get_game_data.map { |k,v| "#{k}=#{v}" }.join(';')
+        score_file.close
+        '----------Score saved!---------'
+    end
+
+    def submit_guess(guess)
+      if (valid_guess?(guess))
+        @attempts -= 1
+        marker = Marker.new(instance_variable_get(:@secret_code), guess)
+        @marking_guess = marker.marking_guess
+        result_after_marking
+      else
+        'Your code is not valid!'
+      end
+    end
+
+    def valid_guess?(guess)
+      return false unless !!guess
+      return false unless guess.length == 4
+      return false unless /[1-6]+/=~guess
+      true
+    end
+
+    def result_after_marking
+      message_for_user = {}
+      message_for_user[:marking_guess] = @marking_guess
+      message_for_user[:game_end] = '----------Congratulations! You win!----------' if @marking_guess == '++++'
+      if @attempts.zero?
+        message_for_user[:secret_code] = 'Secret code is ' + @secret_code.to_s
+        message_for_user[:game_end] = '----------Sorry, but you lose :(----------'
+      end
+      message_for_user[:commands] = 'You can use Hint. Enter it if you want.' if @hint
+      message_for_user[:commands] = 'You can Exit/Restart/Save. Enter the desired command:' if !!message_for_user[:game_end]
+      unless !!message_for_user[:game_end]
+        message_for_user[:attempts] = 'You have ' + @attempts.to_s + ' attempts.'
+        message_for_user[:next_step] = 'Enter a guess:'
+      end
+      message_for_user
+    end
```

Comment:
1. Method is long
2. Remove extra data
3. Remove extra if/else
4. Try service here ?